packages/suite/src/actions/suite/copyAddressActions.ts

```diff
@@ -0,0 +1,25 @@
+import { AddressType } from '@suite-common/wallet-types';
+
+import { copyToClipboard } from '@trezor/dom-utils';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import { openModal } from 'src/actions/suite/modalActions';
+import { Dispatch } from 'redux';
+
+export const copyAddressAction =
+    (address: string, addressType: AddressType, shouldShowCopyAddressModal: boolean) =>
+    (dispatch: Dispatch) => {
+        if (shouldShowCopyAddressModal) {
+            dispatch(
+                openModal({
+                    type: 'copy-address',
+                    addressType,
+                    address,
+                }),
+            );
+        } else {
+            const result = copyToClipboard(address);
+            if (typeof result !== 'string') {
```

Comment:
the condition is unclear at first glance, but a named variable makes it self-explanatory:
```suggestion
            const isError = typeof result === 'string'
            if (!isError) {
```